validation/src/main/scala/hmda/validation/api/ValidationApi.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.api
+
+import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.engine.ValidationError
+import scalaz._
+import scalaz.Scalaz._
+
+trait ValidationApi {
+
+  def convertResult[A](input: A, result: Result, ruleName: String): ValidationNel[ValidationError, A] = {
```

Comment:
Thanks for your response. In the spirit of our team discussion yesterday, I wanted to give you feedback: Please assume a higher level of scala knowledge from me, since I've taken a course in it. In this case, I do already know what type parameters are and that `T` is frequently used for them. I do appreciate your willingness to explain! This will be a great way for me to learn concepts in the future.

Maybe I should have asked, what specific types do you expect `T` to be a stand-in for? Or, in other words, where else will `ValidationApi` be used? Will we be validating anything besides `TransmittalSheet`s and `LoanApplicationRecord`s?